pkg/arvo/sur/group.hoon

```diff
@@ -0,0 +1,93 @@
+/-  *resource
+::
+^?
+|%
+::  $groups: a mapping from group-ids to groups
+::
++$  groups  (map resource group)
+::  $group-tag: an identifier used by groups
+::
+::    These tags should have precise semantics, as they are shared across all
+::    apps.
+::
++$  group-tag  ?(role-tag)
+::  $tag: an identifier used to identify a subset of members
+::
+::   Tags may be used and recognised differently across apps.
+::   for example, you could use tags like `%author`, `%bot`, `%flagged`...
+::
++$  tag  $@(group-tag [app=term tag=term])
+::  $role-tag: a kind of $group-tag that identifies a privileged user
+::
+::    These roles are
+::    %admin: Administrator, can do everything except delete the group
+::    %moderator: Moderator, can add/remove/ban users
+::    %janitor: Has no special meaning inside group-store,
```

Comment:
I think that roles should be %group-store specific, and tags should be used within apps for everything else. I think it's best to remove the %janitor option.

Reply:
Roles won't really be group-store specific though. For example, you can imagine chat would want to restrict pinning a message to certain roles. Having a new tag for each permission on each app would lead to a combinatorial explosion of tags, most of which would probably be duplicating %group-store's roles anyway.